src/main/java/io/appium/java_client/AppiumFluentWait.java

```diff
@@ -222,32 +245,46 @@ public <V> V until(Function<? super T, V> isTrue) {
             // Check the timeout after evaluating the function to ensure conditions
             // with a zero timeout can succeed.
             if (end.isBefore(getClock().instant())) {
-                String message = getMessageSupplier() != null ? getMessageSupplier().get() : null;
-
-                String timeoutMessage = String.format(
-                        "Expected condition failed: %s (tried for %d second(s) with %s interval)",
-                        message == null ? "waiting for " + isTrue : message,
-                        getTimeout().getSeconds(), getInterval());
-                throw timeoutException(timeoutMessage, lastException);
+                handleTimeoutException(lastException, isTrue);
             }
 
-            try {
-                Duration interval = getInterval();
-                if (pollingStrategy != null) {
-                    final IterationInfo info = new IterationInfo(iterationNumber,
-                            Duration.between(start, getClock().instant()), getTimeout(),
-                            interval);
-                    interval = pollingStrategy.apply(info);
-                }
-                getSleeper().sleep(interval);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                throw new WebDriverException(e);
-            }
+            Duration interval = getIntervalWithPollingStrategy(start, iterationNumber);
+            sleepWaitFor(interval);
```

Comment:
rename to `sleepInterruptibly`

Reply:
The correct way should be `sleepUninterruptible`

Reply:
Done

Reply:
why sleepUninterruptible ? Uninterruptible is when the slieep ignored Interruptions, but here we do break it when such exception is received and wrap it into WebDriverException afterwards

Reply:
Ah ok. Now I see your point. Thanks for the suggestion.

Reply:
Done